File: src/iblogstreet/creativepatterns/singleton/SingletonPatternLazy.java
package iblogstreet.creativepatterns.singleton;

/**
 * @author Armyone
 * @date 2019/2/13 16:38
 * @desc 创建型模式-单例模式-懒汉式
 */

public class SingletonPatternLazy {
    private static SingletonPatternLazy instance;

    private SingletonPatternLazy() {
    }

    public static SingletonPatternLazy getInstance() {
        if (instance == null) {
            instance = new SingletonPatternLazy();
        }
        return instance;
    }
}

File: src/kotlin/creativepatterns/SingletonPatternLazy.kt
//package kotlin.creativepatterns
//
///**
// * @author Armyone
// * @date 2019/2/13 17:20
// * @desc
// */
//class SingletonPatternLazy private constructor() {
//    companion object {
//        private var instance: SingletonPatternLazy? = null
//            get() {
//                if (field == null) {
//                    field = SingletonPatternLazy()
//                }
//                return field;
//            }
//
//        fun get(): SingletonPatternLazy {
//            return instance!!
//        }
//    }
//}